dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/context/MultiThreadedContextServlet.java

```diff
@@ -0,0 +1,160 @@
+import jakarta.annotation.Resource;
+import jakarta.enterprise.concurrent.ManagedExecutorService;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.servlet.annotation.WebServlet;
+
+@SuppressWarnings("serial")
+@WebServlet("/context")
+@ApplicationScoped
+public class MultiThreadedContextServlet extends FATServlet {
+
+    private static final String MY_KEY_1 = "myKey1";
+    private static final String MY_KEY_2 = "myKey2";
+    private static final String MY_VALUE_1 = "myValue1";
+    private static final String MY_VALUE_2 = "myValue2";
+    private static final String MY_VALUE_3 = "myValue3";
+
+    @Resource
+    ManagedExecutorService managedExecutorService;
+
+    @Test
+    public void testContextTaskWrapping() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage(MY_KEY_1, MY_VALUE_1);
+
+        // execute my async sub-task
+        ExecutorService executorService = Context.taskWrapping(managedExecutorService);
+        Future<String> future = executorService.submit(new GetBaggageValueCallable(MY_KEY_1));
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        assertEquals(MY_VALUE_1, baggageValue);
+
+        //check that the baggage was not modified by the sub-task
+        assertBaggage(MY_KEY_1, MY_VALUE_1);
+    }
+
+    @Test
+    public void testCurrentContextWrap() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage(MY_KEY_1, MY_VALUE_1);
+
+        // execute my async sub-task
+        Context context = Context.current();
+        ExecutorService executorService = context.wrap(managedExecutorService);
+        Future<String> future = executorService.submit(new GetBaggageValueCallable(MY_KEY_1));
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        assertEquals(MY_VALUE_1, baggageValue);
+
+        //check that the baggage was not modified by the sub-task
+        assertBaggage(MY_KEY_1, MY_VALUE_1);
+    }
+
+    @Test
+    public void testNewContextWrap() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage(MY_KEY_1, MY_VALUE_1);
+
+        //create a new context with a different value in (should not contain MY_KEY_1)
+        Context context = Context.root().with(ContextKey.named(MY_KEY_2), MY_VALUE_2);
+        // execute my async sub-task
+        ExecutorService executorService = context.wrap(managedExecutorService);
+        Future<String> future = executorService.submit(new GetBaggageValueCallable(MY_KEY_1));
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        //check that the value for MY_KEY_1 was null
+        assertEquals(null, baggageValue);
+
+        //check that the baggage of the current context was not modified by the sub-task
+        assertBaggage(MY_KEY_1, MY_VALUE_1);
+    }
```

Comment:
This test (and the following one) doesn't actually do anything that checks that the specified context is propagated. The asserted behaviour would occur if no context was propagated (though it does check that the current context is _not_ propagated).